Assets/Source/Components/Player/Base/PlayerPartComponentBase.cs

```diff
@@ -0,0 +1,18 @@
+﻿using Assets.Source.Components.Base;
+using Assets.Source.Components.Reactor.Interfaces;
+using UnityEngine;
+
+namespace Assets.Source.Components.Player.Base
+{
+    public abstract class PlayerPartComponentBase : ComponentBase, IProjectileReactor
+    {
+        public abstract int BaseDamage { get; }
+
+
+
+        public void ReactToProjectileHit(Collision2D collision, int baseDamage)
+        {
+            throw new System.NotImplementedException();
+        }
```

Comment:
So i believe i just added something very similar to this idea with my enemy logic because i needed the enemy to behave as a "projectile" so eventually we will need to use this for enemies too.